tests/test_seerpy.py

```diff
@@ -18,61 +18,57 @@
 # not really a problem for these test classes
 # pylint:disable=too-few-public-methods
 
+# so that usage of fixtures doesn't get flagged
+# pylint: disable=redefined-outer-name
+
 TEST_DATA_DIR = pathlib.Path(__file__).parent / "test_data"
 
 DEFAULT_CONNECTION_PARAMS = {'url': '.'}
 
 
-@mock.patch('seerpy.seerpy.SeerAuth', autospec=True)
-class TestSeerConnect:
-    def test_success(self, seer_auth):
-        seer_auth.return_value.cookie = {DEFAULT_COOKIE_KEY: "cookie"}
+@pytest.fixture
+def seer_connect():
+    return SeerConnect(seer_auth=auth.BaseAuth(api_url=''))
 
-        result = SeerConnect()
+
+class TestSeerConnect:
+    def test_success(self):
+        result = SeerConnect(seer_auth=auth.BaseAuth(api_url=''))
 
         assert result.graphql_client
```

Comment:
I'm not familiar with pytest fixtures, but is this equivalent to

```suggestion
        assert seer_connect.graphql_client
```

?

Reply:
No, that test doesn't use a fixture, the fixture just happens to have the same result as this test.
I used BaseAuth for both because creating one doesn't have any side effects that I don't care about, so I don't have to mock anything to simplify it.
This test is checking the init, that if you pass a valid auth, then everything works, and it checks that by checking that the graphql_client has a a value.
The fixture is used in other tests that aren't testing init, but that want a setup SeerConnect to test other methods on it. If you check most other test methods, they have a final parameter named seer_connect, that doesn't correspond to any passed in mocks. If there is a missing parameter, Mock looks for a fixture of the same name and passes the return value of that function as the parameter.
That replaces a lot of the patches and mock value setting that was there previously.

Reply:
BTW if you ever come across the error "No fixture named X" when you've got the wrong number of parameters for a test method compared to the number of patches, now you why you get that message instead of something about patches or mocks.

Reply:
Ah ok, thanks for the explanation!